rpc/internal/buildlogger_service.go

```diff
@@ -66,6 +67,24 @@ func (s *buildloggerService) StreamLog(stream Buildlogger_StreamLogServer) error
 	return nil
 }
 
+// CloseLog "closes out" a buildlogger log by setting the completed at
+// timestamp and the exit code. This should be the last rcp call made on a log.
 func (s *buildloggerService) CloseLog(ctx context.Context, info *LogEndInfo) (*BuildloggerResponse, error) {
-	return nil, nil
+	log := &model.Log{ID: info.LogId}
+	log.Setup(s.env)
+	if err := log.Find(); err != nil {
+		return nil, errors.Wrapf(err, "problem finding log record for '%s'", info.LogId)
+	}
+
+	completedAt := time.Now()
+	if info.CompletedAt != nil {
+		var err error
+		completedAt, err = ptypes.Timestamp(info.CompletedAt)
+		if err != nil {
+			return nil, errors.Wrap(err, "problem converting completed_at timestamp")
+		}
+	}
+
+	log.Setup(s.env)
+	return &BuildloggerResponse{LogId: log.ID}, errors.Wrapf(log.CloseLog(completedAt, int(info.ExitCode)), "problem closing log with id %s", log.ID)
```

Comment:
It might be the case that you want to avoid returning errors from these functions in all cases (e.g. remember the stuff about error codes and what not? Return errors might not be big enough

Reply:
gRPC will return these errors with an unknown error code: `rpc error: code = Unknown desc = problem finding log record for '3d3dca54db40c6b0ca6a770f3376da833dcdd797': cannot find with a nil environment`. I can use the gRPC status package to set the rpc code, but then I lose the ability to wrap error messages. What do you think?

Reply:
Either way is probably fine. I don't think we get a lot out of wrapping in this context.

Alternatively, (and for future work) we could see if there was a way to have the middleware take care of wrapping/unwrapping errors.